### PR TITLE
[DO NOT MERGE] K8s version

### PR DESCRIPTION
**Pull request by davidbalazs93**
The approach is to have all dynamic configuration (e.g. `read-urls`, `read-credentials`) stored into configMaps and secrets and mount them into volumes (json files) inside the Docker container, then check them every X minutes and reload them in case changes are detected.

### CODE REVIEW

**Comment by tamas-molnar** [comment]
Maybe we could go with a simpler solution, instead of watchers to have an endpoint to call to refresh the configs, or run the refresh periodically?

File: environments.go [hunk 1]
```diff
@@ -1,181 +1,181 @@
...
-			}
-		}
-		if s3Url == "" {
-			infoLogger.Printf("No S3 url supplied for access to environment %v", name)
-		}
+	return true
+}
 
-		environments[name] = Environment{name, readUrl, s3Url, username, password}
+func computeMD5Hash(fileName string) ([]byte, error) {
```
Comment (tamas-molnar) [inline environments.go]: Can this hash be a string? Would be easier to work with.

File: environments.go [hunk 2]
```diff
@@ -1,181 +1,181 @@
...
+		return fmt.Errorf("Cannot update envs. Error was: %s", err)
+	}
 
 	return nil
 }
 
-func redefineEnvironments(environments map[string]Environment) error {
-	etcdReadEnvResp, err := etcdKeysAPI.Get(context.Background(), *readEnvKey, &etcd.GetOptions{Sort: true})
+func isFileChanged(fileName string) (bool, error) {
+	currentHashing, err := computeMD5Hash(fileName)
```
Comment (tamas-molnar) [inline environments.go]: Maybe `currentHash` is enough? Why the `hashing`?

File: app.go [hunk 3]
```diff
@@ -120,8 +124,19 @@ func main() {
 		return
 	}
 
-	go DiscoverEnvironmentsAndValidators(etcdPeers, etcdReadEnvKey, etcdCredKey, etcdS3EnvKey, etcdValidatorCredKey, environments)
+	err = updateEnvsIfChanged(*envsFileName, *envCredentialsFileName)
+	if err != nil {
+		errorLogger.Printf("Cannot load envs config, error was: [%v]", err)
+		return
```
Comment (tamas-molnar) [inline app.go]: Which exit code does this use? Would be useful if it used an error code so fleet knows to restart it after the configured period.

File: environments.go (hunk 2 above)
Comment (davidbalazs93) [inline environments.go]: :+1:

File: app.go (hunk 3 above)
Comment (davidbalazs93) [inline app.go]: :+1:

File: environments.go (hunk 1 above)
Comment (davidbalazs93) [inline environments.go]: I tried to avoid the encodeToString step, but it doesn't seem to bring any significant value.

**Comment by davidbalazs93** [comment]
further development here: https://github.com/Financial-Times/publish-availability-monitor/pull/145

**Pull request closed without merging**